### PR TITLE
update ui and add button to open explore generators from yeoman ui

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: backend/src/exploregens.ts [hunk 1]
```diff
@@ -119,6 +119,7 @@ export class ExploreGens {
     }
 
     private async getFilteredGenerators(query = this.EMPTY, author = this.EMPTY) {
+        query = query || this.EMPTY;
```
Comment (tomer-epstein) [inline backend/src/exploregens.ts:122]: not needed. var already set at function call.

File: backend/src/panels/ExploreGensPanel.ts [hunk 2]
```diff
@@ -14,9 +14,13 @@ export class ExploreGensPanel extends AbstractWebviewPanel {
     }
 
     public exploreGenerators() {
-        this.disposeWebviewPanel();
-        const webViewPanel = this.createWebviewPanel();
-        this.setWebviewPanel(webViewPanel);
+        if (this.webViewPanel) {
+			this.webViewPanel.reveal();
+		} else {
```
Comment (tomer-epstein) [inline backend/src/panels/ExploreGensPanel.ts:19]: 2 X shift+tab

File: backend/src/panels/YeomanUIPanel.ts [hunk 3]
```diff
@@ -20,9 +20,13 @@ export class YeomanUIPanel extends AbstractWebviewPanel {
...
-		this.disposeWebviewPanel();
-		const webViewPanel = this.createWebviewPanel();
-		this.setWebviewPanel(webViewPanel, uiOptions);
+		if (this.webViewPanel && _.isEmpty(uiOptions)) {
+			this.webViewPanel.reveal();
+		} else {
+			this.disposeWebviewPanel();
+			const webViewPanel = this.createWebviewPanel();
+			this.setWebviewPanel(webViewPanel, uiOptions);
+		}
```
Comment (tomer-epstein) [inline backend/src/panels/YeomanUIPanel.ts:29]: 👍

File: backend/src/exploregens.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/exploregens.ts:122]: It does not work when null is passed

Comment (slavik-lvovsky) [reply backend/src/exploregens.ts:122]: i have also changed the "author" parameter

File: backend/src/panels/ExploreGensPanel.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply backend/src/panels/ExploreGensPanel.ts]: Done
Strange in vscode it looks good

File: backend/src/panels/YeomanUIPanel.ts (hunk 3 above)
Comment (tomer-epstein) [reply backend/src/panels/YeomanUIPanel.ts:29]: would be best taking this block into AbstractWebviewPane. call it loadWebviewPanel(uiOptions);

**Pull request merged**